### PR TITLE
ML refactoring

### PR DESCRIPTION
**Pull request by babenek**
## Description

Please include a summary of the change and which is fixed.

- ML refactoring
- add line, variable lstm layers
- retrain the model

ml_model

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTest
- [x] Benchmark

### CODE REVIEW

File: experiment/src/data_loader.py [hunk 1]
```diff
@@ -93,23 +110,95 @@ def read_metadata(meta_dir: str, split="CredData/") -> Dict[identifier, Dict]:
...
+    #             "value": None,
+    #             "value_start": i["ValueStart"],  # remove
+    #             "value_end": i["ValueEnd"],  # remove
+    #             "variable": None,  # ???
+    #             'RuleName': (x for x in i["Category"].split(':') if x in line_index_set),
+    #             'GroundTruth': 'T',
+    #             'ext': Util.get_extension(i["FilePath"]),
+    #             'type': i["FilePath"].split('/')[-2]
+    #         }
+    #         assert markup_data.keys() == reference_line_data.keys(), reference_line_data.keys()
```
Comment (csh519) [inline experiment/src/data_loader.py:199]: It seems old code left.. Please remove if you don't need it.

File: tests/test_main.py [hunk 2]
```diff
@@ -798,7 +798,6 @@ def test_param_n(self) -> None:
     def test_param_p(self) -> None:
         # internal parametrized tests for quick debug
         items = [  #
-            ("prod.py", b"secret_api_key='Ah\\tga%$FiQ@Ei8'", "secret_api_key", "Ah\\tga%$FiQ@Ei8"),  #
```
Comment (csh519) [inline tests/test_main.py:801]: It seems this PR makes some FPs...
Why those cases are removed?

Comment (babenek) [reply tests/test_main.py:801]: ``\t``  sign did not appear in train set, so the sequence gives low ml probability. I'll update the sample instead removing

**Review by csh519 — COMMENTED** [review]
Thank you for refactor the ML model.
I have one suggestion about typo, please check below.

**Review by csh519 — APPROVED** [review]
ML model has been refactored.

LGTM :+1:

**Review by xDizzix — APPROVED** [review]

**Pull request merged**